File: aoc2022/01-calorie-counting/calorie_counting.c
#include <stdio.h>
#define STB_DS_IMPLEMENTATION
#include "stb_ds.h"
#define SV_IMPLEMENTATION
#include "sv.h"

void part_one(void) {

}

void part_two(void) {

}

int main(int argc, char **argv) {
    if (argc < 2) {
        printf("Must supply filepath to puzzle input as an argument\n");
        exit(1);
    }

    String_View filedata = sv_read_entire_file(argv[1]);

    fread

    String line = chop_by_delimiter(filedata, '\n');

    isspace

    return 0;
}

File: aoc2022/01-calorie-counting/calorie_counting.py
if __name__ == "__main__":
    counts = []

    count = 0
    max = 0

    with open("input", "r") as f:
        end = False
        while 1:
            line = f.readline().strip()
            if not line:
                if count > max:
                    max = count
                counts.append(count)
                count = 0
                if end:
                    break
                end = True
                continue
            end = False
            count += int(line)
            
    print(f"part_one: {max}")

    sum = 0
    for c in sorted(counts)[-3:]:
        sum += c

    print(f"part_two: {sum}")

